systemtest/src/test/java/io/strimzi/systemtest/upgrade/OlmUpgradeIsolatedST.java

```diff
@@ -55,117 +54,105 @@ public class OlmUpgradeIsolatedST extends AbstractUpgradeST {
 
     private static final Logger LOGGER = LogManager.getLogger(OlmUpgradeIsolatedST.class);
 
-    private final String producerName = "producer";
-    private final String consumerName = "consumer";
-    private final String topicUpgradeName = "topic-upgrade";
-    // clusterName has to be same as cluster name in examples
-    private final String clusterName = "my-cluster";
-    private final int messageUpgradeCount =  600;
-    private KafkaClients kafkaBasicClientJob;
-    private KafkaTopic kafkaUpgradeTopic;
-
     @Test
     void testStrimziUpgrade(ExtensionContext extensionContext) throws IOException {
-        UpgradeDowngradeDatalist upgradeDowngradeDatalist = new UpgradeDowngradeDatalist();
-        UpgradeDowngradeData latestUpgradeData = upgradeDowngradeDatalist.getUpgradeData(upgradeDowngradeDatalist.getUpgradeDataSize() - 1);
-
-        List<TestKafkaVersion> testKafkaVersions = TestKafkaVersion.getSupportedKafkaVersions();
-        TestKafkaVersion testKafkaVersion = testKafkaVersions.get(testKafkaVersions.size() - 1);
-
-        // Generate procedures and data for OLM upgrade
-        // This is needed, because for upgrade it's generated by method which generated data for parametrized test
-        UpgradeKafkaVersion procedures = new UpgradeKafkaVersion(testKafkaVersion.version());
-
-        latestUpgradeData.setProcedures(procedures);
-        latestUpgradeData.setToVersion(Environment.OLM_OPERATOR_LATEST_RELEASE_VERSION);
+        // Fetch upgrade data from yaml and parse them to class
+        OlmVersionModificationData olmUpgradeData = new VersionModificationDataLoader(ModificationType.OLM_UPGRADE).getOlmUpgradeData();
```

Comment:
Yes, i thought it would be better if you wanted to call getData multiple times, the initial definition and parsing from yamlnwould happen again

Reply:
Makes sense, thanks for explanation :) I thought about option to load it in `BeforeAll` and then just do the `get` calls -> that means it could be singleton and if it would be already initialized it would use that data.

Reply:
Static variable inside each class instead of singleton will be put in use.